src/main/java/manageezpz/model/ModelManager.java

```diff
@@ -219,29 +219,19 @@ public void updateFilteredTaskList(Predicate<Task> predicate) {
     }
 
     @Override
-    public boolean hasEvent(Event event) {
-        requireNonNull(event);
-        return addressBook.hasEvent(event);
-    }
-
-    @Override
-    public boolean hasDeadline(Deadline deadline) {
-        requireNonNull(deadline);
-        return addressBook.hasDeadline(deadline);
-    }
-
-    @Override
-    public boolean hasTodo(Todo todo) {
-        requireNonNull(todo);
-        return addressBook.hasTodo(todo);
+    public boolean hasTask(Task task) {
+        requireNonNull(task);
+        return addressBook.hasTask(task);
```

Comment:
Good to remove other methods - hasEvent, hasDeadline and hasTodo, since hasTask would be sufficient due to inheritance